python/libcuopt/CMakeLists.txt

```diff
@@ -38,23 +38,140 @@ endif()
 unset(cuopt_FOUND)
 
 
+# Set up Boost variables globally before any FetchContent operations
+# This ensures any subproject that tries to find_package(Boost) will find our setup
+set(Boost_FOUND TRUE CACHE BOOL "Boost found via FetchContent" FORCE)
+set(Boost_IOSTREAMS_FOUND TRUE CACHE BOOL "Boost iostreams found" FORCE)
+set(Boost_PROGRAM_OPTIONS_FOUND TRUE CACHE BOOL "Boost program_options found" FORCE)
+set(Boost_SERIALIZATION_FOUND TRUE CACHE BOOL "Boost serialization found" FORCE)
+set(Boost_FILESYSTEM_FOUND TRUE CACHE BOOL "Boost filesystem found" FORCE)
+set(Boost_SYSTEM_FOUND TRUE CACHE BOOL "Boost system found" FORCE)
+set(Boost_THREAD_FOUND TRUE CACHE BOOL "Boost thread found" FORCE)
+set(Boost_VERSION "1.85.0" CACHE STRING "Boost version" FORCE)
+set(Boost_VERSION_STRING "1.85.0" CACHE STRING "Boost version string" FORCE)
+set(PAPILO_HAVE_BOOST_IOSTREAMS 1 CACHE STRING "Papilo boost iostreams" FORCE)
+set(PAPILO_COMMAND_LINE_AVAILABLE 1 CACHE STRING "Papilo command line available" FORCE)
+set(PAPILO_SERIALIZATION_AVAILABLE 1 CACHE STRING "Papilo serialization available" FORCE)
+
 include(FetchContent)
+
+# Fetch dependencies
 FetchContent_Declare(
   argparse
   GIT_REPOSITORY https://github.com/p-ranav/argparse.git
   GIT_TAG v3.2
 )
 FetchContent_MakeAvailable(argparse)
 
+FetchContent_Declare(
+  boost
+  GIT_REPOSITORY https://github.com/boostorg/boost.git
+  GIT_TAG boost-1.85.0
+)
+FetchContent_MakeAvailable(boost)
+
+# Path to boost root
+set(BOOST_ROOT "${boost_SOURCE_DIR}")
+
+# Only do this the first time Boost is fetched
+if(NOT EXISTS "${boost_SOURCE_DIR}/boost/version.hpp")
+  message(STATUS "Running Boost bootstrap and b2 headers...")
+
+  execute_process(
+    COMMAND ./bootstrap.sh
+    WORKING_DIRECTORY ${boost_SOURCE_DIR}
+    RESULT_VARIABLE BOOTSTRAP_RESULT
+  )
+  if(NOT BOOTSTRAP_RESULT EQUAL 0)
+    message(FATAL_ERROR "Boost bootstrap.sh failed with code ${BOOTSTRAP_RESULT}")
+  endif()
+
+  execute_process(
+    COMMAND ./b2 headers
+    WORKING_DIRECTORY ${boost_SOURCE_DIR}
+    RESULT_VARIABLE B2_RESULT
+  )
+  if(NOT B2_RESULT EQUAL 0)
+    message(FATAL_ERROR "Boost b2 headers failed with code ${B2_RESULT}")
+  endif()
+ 
+  execute_process(
+    COMMAND ./b2
+      link=shared
+      variant=release
+      --with-regex
+      --with-serialization
+      --with-program_options
+      --with-iostreams
+      threading=multi
+      cxxflags=-fPIC
+    WORKING_DIRECTORY ${boost_SOURCE_DIR}
+    RESULT_VARIABLE B2_BUILD_RESULT
+  )
+
+  if(NOT B2_BUILD_RESULT EQUAL 0)
+    message(FATAL_ERROR "Boost b2 library build failed with code ${B2_BUILD_RESULT}")
+  endif()
+
+endif()
+
+# --- Add include path manually ---
+include_directories(${boost_SOURCE_DIR})
+
+list(APPEND CMAKE_PREFIX_PATH "${boost_SOURCE_DIR}/stage/lib/cmake")
+
+list(APPEND CMAKE_PREFIX_PATH
+  "/repo/python/libcuopt/build/py3-none-linux_x86_64/_deps/boost-src/stage/lib/cmake"
```

Comment:
You can't use absolute paths like this. You need to compute this based on `CMAKE_BINARY_DIR`